static/js/components/UploadSpectrum.jsx

```diff
@@ -313,7 +316,10 @@ const UploadSpectrumForm = ({ route }) => {
           type: "integer",
           anyOf: users?.map((user) => ({
             enum: [user.id],
-            title: user.username,
+            title:
+              user?.first_name && user?.last_name
+                ? `${user.username} (${user.first_name} ${user.last_name})`
+                : `${user.username}`,
```

Comment:
```suggestion
            title: `${user.username} (${first ? first + " " : ""}${last ? last : ""})`,
```

Reply:
Same below.

Otherwise, if either first or last name is missing, the real name won't render.

Reply:
fixed!